src/OSPSuite.UI/Views/Importer/ColumnMappingView.cs

```diff
@@ -74,29 +87,59 @@ private void queryDisplayText(QueryDisplayTextEventArgs e)
       {
          var withValueOrigin = _gridViewBinder.FocusedElement;
          if (withValueOrigin == null) return;
-         e.DisplayText = withValueOrigin.MappingName;
+         switch (withValueOrigin.Source)
+         {
+            case MappingDataFormatParameter _:
+               e.DisplayText = withValueOrigin.MappingName;
+               break;
+            case MetaDataFormatParameter md:
+               e.DisplayText = md.ColumnName;
+               break;
+         }
       }
 
       private RepositoryItem repositoryItemPopupContainerEdit(ColumnMappingDTO model)
       {
-         if (!(model.Source is MappingDataFormatParameter))
-            return _disabledPopupContainerEdit;
-
-         _presenter.SetSubEditorSettings(model);
-         return _repositoryItemPopupContainerEdit;
+         switch (model.Source)
+         {
+            case MappingDataFormatParameter _:
+               _presenter.SetSubEditorSettingsForMapping(model);
+               return _repositoryMappingPopupContainerEdit;
+            case MetaDataFormatParameter md:
+               var metaDataCategory = _metaDataCategories.FirstOrDefault(x => x.Name == model.MappingName);
```

Comment:
THis logic should be in presenter, not in view

Reply:
Done